src/open_inwoner/react/components/SelectComboBox/SelectComboBox.tsx

```diff
@@ -0,0 +1,388 @@
+      }
+    }
+  }, [selectedId, name])
+
+  return (
+    <div
+      className="utrecht-form-field utrecht-form-field--text"
+      ref={containerRef}
+    >
+      <label htmlFor={id} className="utrecht-form-label">
+        {label}
+      </label>
+
+      <div className="oip-combobox">
+        <input
+          ref={inputRef}
+          id={id}
+          className="utrecht-textbox oip-combobox__input"
+          type="text"
+          role="combobox"
+          value={query}
+          onChange={handleInputChange}
+          onFocus={() => {
+            setIsOpen(true)
+            setFocusedIndex(0)
+          }}
+          onKeyDown={handleKeyDown}
+          aria-expanded={isOpen}
+          aria-controls={`${id}-listbox`}
+          placeholder={intl.formatMessage({
+            id: 'combobox.placeholder',
+            description: 'Placeholder text for branch search input',
+            defaultMessage: 'Vul naam, adres of vestigingsnummer in...',
+          })}
+          autoComplete="off"
+        />
+
+        {/* TODO: Replace with NLDS Buttons once available */}
+        {hasText ? (
+          <button
+            type="button"
+            className="button button__icon-close button--textless button--icon oip-combobox__icon"
+            onClick={handleClearQuery}
+            aria-label={intl.formatMessage({
+              id: 'combobox.clear',
+              description: 'Clear search button label',
+              defaultMessage: 'Wissen',
+            })}
+          >
+            <MaterialIcon name="close" />
+          </button>
+        ) : (
+          <button
+            type="button"
+            className="button button--textless button--icon oip-combobox__icon"
+            onClick={handleToggleDropdown}
+            aria-label={intl.formatMessage({
+              id: 'combobox.toggle',
+              description: 'Toggle dropdown button label',
+              defaultMessage: 'Toggle dropdown',
+            })}
+            aria-expanded={isOpen}
+            aria-controls={`${id}-listbox`}
+          >
+            <MaterialIcon name="keyboard_arrow_down" />
+          </button>
+        )}
+
+        {isOpen && filtered.length > 0 && (
+          <div
+            className="utrecht-listbox oip-combobox__popover oip-combobox__popover--block-end"
+            role="listbox"
+            id={`${id}-listbox`}
+            aria-label={label}
+          >
+            <ul className="utrecht-listbox__list" ref={listRef}>
+              {filtered.map((item, i) => (
+                <li
+                  key={item.id}
+                  role="option"
+                  aria-selected={selectedId === item.id}
+                  className={clsx(
+                    'utrecht-listbox__option',
+                    selectedId === item.id &&
+                      'utrecht-listbox__option--selected',
+                    focusedIndex === i && 'utrecht-listbox__option--active'
+                  )}
```

Comment:
With clsx you could do something like this
```suggestion
                  className={clsx(
                    'utrecht-listbox__option', {
                      ['utrecht-listbox__option--selected']: selectedId === item.id,
                      ['utrecht-listbox__option--active']: focusedIndex === i,
                    }
                  )}
```